src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs

```diff
@@ -446,14 +452,20 @@ private void UpdateRunAnalyzers()
         ChangeProjectProperty(ref _runAnalyzers, runAnalyzers, s => s.WithRunAnalyzers(Id, runAnalyzers));
     }
 
+    internal bool HasSdkCodeStyleAnalyzers
```

Comment:
Since this can be calculated from `_unmappedProjectAnalyzerPaths` it'd be simpler to have:

```
private bool? _lazyHasSdkCodeStyleAnalyzers;

internal bool HasSdkCodeStyleAnalyzers
 => _lazyHasSdkCodeStyleAnalyzers ??= _unmappedProjectAnalyzerPaths.Any(IsSdkCodeStyleAnalyzer);
```

```

Reply:
That is, if we even need to cache the value...